### PR TITLE
Minor improvements to quicklook, coadd2d as well as a bug fix

### PR DESCRIPTION
**Pull request by jhennawi**
This PR is preliminary and includes changes from the combine refactor PR#1813 so wait until that clear. I'm just posting it now so that @debora-pe can look at the coadd2d changes.

### CODE REVIEW

File: pypeit/coadd2d.py [hunk 1]
```diff
@@ -895,26 +897,29 @@ def snr_report(self, snr_bar, slitid=None):
...
+                the plate scale for this detector
+            offsets_method (str):
+                A string describing the method used to determine the offsets
         Returns:
 
         """
 
         if offsets_method is not None and offsets is not None:
-            msg_string = msgs.newline() + '---------------------------------------------'
+            msg_string = msgs.newline() + '---------------------------------------------------------------------------------'
```
Comment (VascoSch92) [inline pypeit/coadd2d.py:890]: You know that you can also the `format` method provided by python such that you don't have to hard-code the lines? :)

File: pypeit/coadd2d.py [hunk 2]
```diff
@@ -27,6 +27,7 @@
 from pypeit.core import findobj_skymask
 from pypeit.core.wavecal import wvutils
 from pypeit.core import coadd
+from pypeit.core import parse
```
Comment (kbwestfall) [inline pypeit/coadd2d.py:30]: Remove the commented line below this?

File: pypeit/coadd2d.py [hunk 3]
```diff
@@ -835,7 +836,8 @@ def reduce(self, pseudo_dict, show=False, clear_ginga=True, show_peaks=False, sh
         # maskdef stuff
         if parcopy['reduce']['slitmask']['assign_obj'] and slits.maskdef_designtab is not None:
             # Get plate scale
-            platescale = sciImage.detector.platescale * self.spat_samp_fact
+            platescale = parse.parse_binning(sciImage.detector.binning)[1]*sciImage.detector.platescale*self.spat_samp_fact
+            #platescale = sciImage.detector.platescale * self.spat_samp_fact
```
Comment (kbwestfall) [inline pypeit/coadd2d.py]: remove?

File: pypeit/coadd2d.py [hunk 4]
```diff
@@ -895,26 +897,29 @@ def snr_report(self, snr_bar, slitid=None):
...
         Args:
-            offsets:
-            offsets_method:
-
+            offsets (`numpy.ndarray`_)
+                Array of offsets
+            platescale (float):
+                the plate scale for this detector
+            offsets_method (str):
+                A string describing the method used to determine the offsets
```
Comment (kbwestfall) [inline pypeit/coadd2d.py:885]: Thanks for adding the argument descriptions!  Two requests:
 - Can you add some description to `platescale`?  I.e., is it the native pixel platescale?  After binning?  Does it include the up/down scaling (`spat_samp_fact`), etc?
 - If there is no return, you can remove the `Returns` block.

File: pypeit/coadd2d.py (hunk 1 above)
Comment (kbwestfall) [reply pypeit/coadd2d.py:890]: I would instead use the multiply operator; e.g., `msg_string = msgs.newline() + '-'*50`

File: pypeit/coadd2d.py [hunk 5]
```diff
@@ -1126,17 +1131,18 @@ def compute_offsets(self, offsets):
 
         """
         msgs.info('Get Offsets')
+        platescale = parse.parse_binning(self.stack_dict['detectors'][0].binning)[1]*self.stack_dict['detectors'][0].platescale
+        #platescale = self.stack_dict['detectors'][0].platescale
```
Comment (kbwestfall) [inline pypeit/coadd2d.py]: Remove?

We also might want to use a different variable names for the "platescale" values that do and do not include the scale factor, `spat_samp_fact`, to help make the code easier to read.

File: pypeit/par/pypeitpar.py [hunk 6]
```diff
@@ -1517,6 +1517,23 @@ def __init__(self, only_slits=None, exclude_slits=None, offsets=None, spat_toler
...
+        
+
+        defaults['spat_samp_fact'] = 1.0
+        dtypes['spat_samp_fact'] = float
+        descr['spat_samp_fact'] = "Make the spatial sampling finer (``spat_samp_fact`` less" \
+                "than 1.0) or coarser (``spat_samp_fact`` greather than 1.0) by" \
+                "this sampling factor. This basically multiples the 'native'" \
+                "spatial pixel size by ``spat_samp_fact``, i.e. the units of" \
+                "``spat_samp_fact`` are pixels."
+
```
Comment (kbwestfall) [inline pypeit/par/pypeitpar.py:1573]: Nice!

File: pypeit/scripts/ql.py [hunk 7]
```diff
@@ -714,36 +714,40 @@ def get_parser(cls, width=None):
                             help='If standard star observations are automatically detected, '
                                  'ignore those frames.  Otherwise, they are included with the '
                                  'reduction of the science frames.')
-        parser.add_argument('--skip_display', dest='show', default=True, action='store_false',
-                            help='Run the quicklook without displaying any results.')
+        parser.add_argument('--skip_display', dest='skip_display', default=False, action='store_true',
```
Comment (kbwestfall) [inline pypeit/scripts/ql.py]: If the argument name is the same as the flag, you don't need the `dest` keyword.

File: pypeit/scripts/setup_coadd2d.py [hunk 8]
```diff
@@ -178,7 +186,12 @@ def main(args):
                 if args.weights is None else args.weights
         cfg['coadd2d']['spat_toler'] = par['coadd2d']['spat_toler'] \
                 if args.spat_toler is None else args.spat_toler
-
+        cfg['coadd2d']['spec_samp_fact'] = par['coadd2d']['spec_samp_fact'] \
+                if args.spec_samp_fact is None else args.spec_samp_fact
+        cfg['coadd2d']['spat_samp_fact'] = par['coadd2d']['spat_samp_fact'] \
+                if args.spat_samp_fact is None else args.spat_samp_fact
+        
+        # TODO JFH Why are exclude_slits and only_slits set here when they are parameters in the parset?
```
Comment (kbwestfall) [inline pypeit/scripts/setup_coadd2d.py:194]: Good question.  Assuming I was the one that wrote this, my guess is that I was mimicking the fact that `only_lists` and `exclude_slits` are keyword arguments of `CoAdd2D`.  I'm fine with having them removed as explicit keyword arguments and included as part of `cfg` instead.

**Review by kbwestfall — APPROVED** [review]
Looks good.  Mostly minor comments from me.  Please add some short notes in the release doc (`doc/releases/1.16.1dev.rst`) about these updates.  And we should run tests!

File: pypeit/coadd2d.py [hunk 9]
```diff
@@ -1312,7 +1295,7 @@ def __init__(self, spec2d_files, spectrograph, par, det=1, offsets=None, weights
             self.objid_bri, self.slitidx_bri, self.spatid_bri, self.snr_bar_bri = \
                 np.repeat(user_objid, self.nexp), _slitidx_bri, user_slit, None
         elif len(self.stack_dict['specobjs_list']) > 0 and (offsets == 'auto' or weights == 'auto'):
-            self.objid_bri, self.slitidx_bri, self.spatid_bri, self.snr_bar_bri = \
+            self.objid_bri, self.spat_pixpos_bri, self.slitidx_bri, self.spatid_bri, self.snr_bar_bri = \
```
Comment (debora-pe) [inline pypeit/coadd2d.py:1348]: `self.spat_pixpos_bri` should be defined also in the `if` above and the `else` below. Even if they are  just`None`

**Review by debora-pe — APPROVED** [review]
Looks good to me. I just have one comment. Still, let's run the tests.

**Review by kbwestfall — CHANGES_REQUESTED** [review]
Changing my review to "Request changes" until the comments are addressed and we run tests.

File: pypeit/scripts/setup_coadd2d.py (hunk 8 above)
Comment (debora-pe) [reply pypeit/scripts/setup_coadd2d.py:194]: The method below (`CoAdd2D.default_par()`) is creating the default parameters and there is an interplay between exclude_slits, only_slits, and other default params. I think, what we can do is to just merge/add the lines above to the `CoAdd2D.default_par()` method.

File: pypeit/coadd2d.py (hunk 2 above)
Comment (debora-pe) [reply pypeit/coadd2d.py:30]: done

File: pypeit/coadd2d.py (hunk 3 above)
Comment (debora-pe) [reply pypeit/coadd2d.py]: done

File: pypeit/coadd2d.py (hunk 4 above)
Comment (debora-pe) [reply pypeit/coadd2d.py:885]: done

File: pypeit/coadd2d.py (hunk 5 above)
Comment (debora-pe) [reply pypeit/coadd2d.py]: done

File: pypeit/scripts/ql.py (hunk 7 above)
Comment (debora-pe) [reply pypeit/scripts/ql.py]: removed

**Comment by debora-pe** [comment]
Tests pass! Note that the PYTEST UNIT TESTS failures are still due to a problem of my machine with `setup_gui`. The `keck_deimos/900ZD_M_6000` failure was probably a glitch in Google Drive since I re-run it without changing anything and it passed.


```
Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS PASSED  256 passed, 62 warnings in 374.80s (0:06:14) ---
--- PYTEST UNIT TESTS FAILED  7 failed, 145 passed, 168 warnings in 5952.81s (1:39:12) ---
--- PYTEST VET TESTS PASSED  62 passed, 100 warnings in 5457.14s (1:30:57) ---
--- PYPEIT DEVELOPMENT SUITE FAILED 1/247 TESTS  ---
Failed tests:
    keck_deimos/900ZD_M_6000 pypeit
Skipped tests:
Testing Started at 2024-09-18T13:50:53.456748
Testing Completed at 2024-09-19T04:35:23.424708
Total Time: 14:44:29.967960
```

```
./pypeit_test reduce -i keck_deimos -s 900ZD_M_6000
Running reduce tests.
Running tests on the following instruments:
    keck_deimos

Reducing data from keck_deimos for the following setups:
    900ZD_M_6000

 0 passed/ 0 failed/ 0 skipped STARTED keck_deimos/900ZD_M_6000 pypeit
 1 passed/ 0 failed/ 0 skipped PASSED  keck_deimos/900ZD_M_6000 pypeit
--- PYPEIT DEVELOPMENT SUITE PASSED 1/1 TESTS  ---

Test Summary
--------------------------------------------------------
--- PYPEIT DEVELOPMENT SUITE PASSED 1/1 TESTS  ---
Testing Started at 2024-09-19T11:39:00.760448
Testing Completed at 2024-09-19T11:54:02.169846
Total Time: 0:15:01.409398
```

**Comment by debora-pe** [comment]
@kbwestfall this is ready to be merged. Please, take a look and see if you agree with it.

File: pypeit/core/coadd.py [hunk 10]
```diff
@@ -1859,18 +1859,18 @@ def spec_reject_comb(wave_grid, wave_grid_mid, waves_list, fluxes_list, ivars_li
...
-                                                               flux_stack_nat,  ivar_stack_nat, gpm_stack_nat, sn_clip=sn_clip)
-        this_gpms, qdone = pydl.djs_reject(fluxes, flux_stack_nat, outmask=this_gpms,inmask=gpms, invvar=rejivars,
+        rejivars, sigma_corrs, outchi, chigpm = update_errors(
+            fluxes, ivars, this_gpms, flux_stack_nat,  ivar_stack_nat, gpm_stack_nat, sn_clip=sn_clip)
+        this_gpms, qdone = pydl.djs_reject(fluxes, flux_stack_nat, outmask=this_gpms, inmask=gpms, invvar=rejivars,
                                           lower=lower,upper=upper, maxrej=maxrej, sticky=False)
+        #rejivars, sigma_corrs, outchi, chigpm = update_errors(
+        #    fluxes, ivars, this_gpms, flux_stack_nat,  ivar_stack_nat, gpm_stack_nat, sn_clip=sn_clip)
+        #this_gpms, qdone = pydl.djs_reject(fluxes, flux_stack_nat, outmask=this_gpms, inmask=gpms, invvar=rejivars,
+        #                                  lower=lower,upper=upper, maxrej=maxrej, sticky=False)
```
Comment (kbwestfall) [inline pypeit/core/coadd.py]: Remove these commented lines?

File: pypeit/scripts/ql.py [hunk 11]
```diff
@@ -1002,9 +1010,9 @@ def main(args):
             
             # Run the coadding
             coadd2dFile = inputfiles.Coadd2DFile.from_file(coadd_file)
-            CoAdd2DSpec.main(CoAdd2DSpec.parse_args([str(coadd_file),
-                                                     '--spec_samp_fact', str(args.spec_samp_fact),
-                                                     '--spat_samp_fact', str(args.spat_samp_fact)]))
+            CoAdd2DSpec.main(CoAdd2DSpec.parse_args([str(coadd_file)]))
+            #                                         '--spec_samp_fact', str(args.spec_samp_fact),
+            #                                         '--spat_samp_fact', str(args.spat_samp_fact)]))
```
Comment (kbwestfall) [inline pypeit/scripts/ql.py]: remove these commented lines?

File: pypeit/scripts/setup_coadd2d.py (hunk 8 above)
Comment (kbwestfall) [reply pypeit/scripts/setup_coadd2d.py:194]: Do we want to deal with this now?

**Review by kbwestfall — APPROVED** [review]
Just a couple clean-up requests and a question about whether we deal with some keyword arguments now or later.

File: pypeit/core/coadd.py (hunk 10 above)
Comment (debora-pe) [reply pypeit/core/coadd.py]: done

File: pypeit/scripts/ql.py (hunk 11 above)
Comment (debora-pe) [reply pypeit/scripts/ql.py]: done

File: pypeit/scripts/setup_coadd2d.py (hunk 8 above)
Comment (debora-pe) [reply pypeit/scripts/setup_coadd2d.py:194]: No, we'll leave it for another PR.

**Comment by debora-pe** [comment]
I'm merging...unless anybody objects.

**Pull request merged**